app/integration_action.go

```diff
@@ -565,6 +565,32 @@ func (a *App) OpenInteractiveDialog(request model.OpenDialogRequest) *model.AppE
 	return nil
 }
 
+func (a *App) SelectInteractiveDialogOption(request model.DialogSelectOptionRequest) (*model.DialogSelectOptionResponse, *model.AppError) {
+	url := request.URL
+	request.URL = ""
+	request.Type = "dialog_select"
+
+	b, jsonErr := json.Marshal(request)
+	if jsonErr != nil {
+		return nil, model.NewAppError("SelectInteractiveDialogOption", "app.select_interactive_dialog_option.json_error", nil, jsonErr.Error(), http.StatusBadRequest)
+	}
+
+	resp, err := a.DoActionRequest(url, b)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	var response model.DialogSelectOptionResponse
+	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
+		// Don't fail, an empty response is acceptable
```

Comment:
Should we then distinguish between an badly formed response and an empty response?

Reply:
@larkox This function is very hackathon-y, sorry about that. I think this code should probably go into the `mattermost-plugin-apps` repo as it stands anyway? I mainly wanted to get this code available before going OOO next week.

Reply:
Also FWIW much of this code was taken from the function above it